### ISSUE TITLE
A training problem about Global alloc not supported yet

### ISSUE THREAD

**Issue by xiehousen** [facebookresearch/Mask2Former#4]
I created a new running environment for mask2former according to the steps. When I train the COCO dataset, I can train normally, but when I train my dataset, I encounter the following problems. 

![2021-12-09 11-50-45 的屏幕截图](https://user-images.githubusercontent.com/37787619/145331595-eb77045a-763e-46d3-9299-a102f6dc2e50.png)

I've been looking for a solution on Google for a long time, so I'd like to ask if you have any similar problems. Thank you very much for your reply.

**Comment by bowenc0221** [comment]
Does it work if you use `batch_dice_loss` instead of `batch_dice_loss_jit`?

**Comment by xiehousen** [comment]
if I use `batch_dice_loss`, that's worked.

**Comment by haotian-liu** [comment]
Hi @bowenc0221, I met the same issue when training the model on YouTube-VIS dataset (with official code).  Do we need to  turn JIT off or is there some way to fix this RuntimeError issue?  Thanks!

**Comment by bowenc0221** [comment]
> Hi @bowenc0221, I met the same issue when training the model on YouTube-VIS dataset (with official code). Do we need to turn JIT off or is there some way to fix this RuntimeError issue? Thanks!

I have never met this error. I think it is not necessary to use the JIT version of the loss function, so turning it off is the simplest solution. If you really want to fix this error, I would suggest posting it to the PyTorch team for help.

**Comment by haotian-liu** [comment]
> > Hi @bowenc0221, I met the same issue when training the model on YouTube-VIS dataset (with official code). Do we need to turn JIT off or is there some way to fix this RuntimeError issue? Thanks!
> 
> I have never met this error. I think it is not necessary to use the JIT version of the loss function, so turning it off is the simplest solution. If you really want to fix this error, I would suggest posting it to the PyTorch team for help.

Thank you!

**Comment by xiehousen** [comment]
> > > Hi @bowenc0221, I met the same issue when training the model on YouTube-VIS dataset (with official code). Do we need to turn JIT off or is there some way to fix this RuntimeError issue? Thanks!
> > 
> > 
> > I have never met this error. I think it is not necessary to use the JIT version of the loss function, so turning it off is the simplest solution. If you really want to fix this error, I would suggest posting it to the PyTorch team for help.
> 
> Thank you!

I did not use the JIT version, and the The training speed becomes very slow. Do you have this question?

**Comment by ShijieVVu** [comment]
Please check

> I created a new running environment for mask2former according to the steps. When I train the COCO dataset, I can train normally, but when I train my dataset, I encounter the following problems.
> 
> ![2021-12-09 11-50-45 的屏幕截图](https://user-images.githubusercontent.com/37787619/145331595-eb77045a-763e-46d3-9299-a102f6dc2e50.png)
> 
> I've been looking for a solution on Google for a long time, so I'd like to ask if you have any similar problems. Thank you very much for your reply.

If your custom training set includes zero annotation images, this error would show up.

**Comment by ShijieVVu** [comment]
> I created a new running environment for mask2former according to the steps. When I train the COCO dataset, I can train normally, but when I train my dataset, I encounter the following problems.
> 
> ![2021-12-09 11-50-45 的屏幕截图](https://user-images.githubusercontent.com/37787619/145331595-eb77045a-763e-46d3-9299-a102f6dc2e50.png)
> 
> I've been looking for a solution on Google for a long time, so I'd like to ask if you have any similar problems. Thank you very much for your reply.

If your custom training set includes zero annotation images, this error would show up.

**Comment by deeptig84** [comment]
Hi @bowenc0221 [bowenc0221](https://github.com/bowenc0221)

I would like to know how to turn off the batch_dice_loss , is it a config change in the Mask2Former , or I need to go and change the code itself?

**Comment by YellowPig-zp** [comment]
Just to add to the comments with my personal experience regarding the issue. Seems like even if your data doesn't contain empty annotations, the codes would still throw out the same bug(I manually removed all the images/annotations for ade20k that do not have labels and encountered the same bug).

Seems like it also has something to do with the pytorch version. For v1.10 the bug appears, but when I down-grade to 1.9.1, it runs like a charm, and the training time is significantly reduced(from ~20 hours to ~9 hours on 8 V100 GPUs with Swin-T backbone for ADE20K semantic segmentation).

Hope this could help!

**Comment by Jianghanxiao** [comment]
Based on above comments, I also find that this case happens when some images have no gt. Below is my modification, which can somehow reduce the influence. It can still use jit most of the time. 
![image](https://user-images.githubusercontent.com/32015651/162598927-7d9f3875-969b-4f0e-ac2b-65c80c14a464.png)

**Comment by lsnls** [comment]
Hi, @Jianghanxiao , your modification is effective. But I am confused about why `tgt_mask.shape[0] == 0` might appear?

Hope this could be help! Thanks a lot!

**Comment by Jianghanxiao** [comment]
> Hi, @Jianghanxiao , your modification is effective. But I am confused about why `tgt_mask.shape[0] == 0` might appear?
> 
> Hope this could be help! Thanks a lot!

I kind of forget the details. But basically both the image GT and prediction can be empty. In some dataset, some images don't contain any mask annotations.

**Comment by lsnls** [comment]
> > Hi, @Jianghanxiao , your modification is effective. But I am confused about why `tgt_mask.shape[0] == 0` might appear?
> > Hope this could be help! Thanks a lot!
> 
> I kind of forget the details. But basically both the image GT and prediction can be empty. In some dataset, some images don't contain any mask annotations.

Thanks a lot! It helps me a lot!
